l10n_nl_cbs_export/models/cbs_export_file.py

```diff
@@ -0,0 +1,279 @@
+            str(" " * 6) + \
+            str(" " * 5) + \
+            str(datetime.now().strftime("%Y%m%d").ljust(8)) + \
+            str(datetime.now().strftime("%H%M%S").ljust(6)) + \
+            str(company.phone or '').replace(
+                '+31', '0').replace(' ', '')[0:15].ljust(15) + \
+            str(" " * 13) + '\r\n'
+
+        return cbs_export_data
+
+    @staticmethod
+    def _format_footer():
+        cbs_export_data = str('9899') + str(" " * 111) + '\r\n'
+        return cbs_export_data
+
+    @api.multi
+    def _format_lines(self):
+        self.ensure_one()
+
+        line_counter = 1
+        invoice_line_number = 0
+        cbs_export_data = ''
+        invoices = self.account_invoice_ids
+
+        for invoice_line in invoices.mapped('invoice_line_ids'):
+            sign_of_weight = '-'
+            sign_of_invoice_value = '-'
+            if invoice_line.invoice_id.amount_total_signed >= 0:
+                sign_of_invoice_value = '+'
+            if (invoice_line.quantity * invoice_line.product_id.weight) >= 0:
+                sign_of_weight = '+'
+            if invoice_line.price_subtotal < 0:
+                sign_of_invoice_value = '-'
+
+            value = \
+                str(datetime.strptime(
+                    invoice_line.invoice_id.date_invoice, '%Y-%m-%d'
+                ).strftime("%Y%m") or '').ljust(6) + \
+                str('7') + \
+                str(invoice_line.company_id.vat or ''
+                    ).replace(' ', '')[2:].ljust(12) + \
+                str(line_counter).zfill(5) + str(" " * 3) + \
+                str(invoice_line.invoice_id.partner_id.country_id.code or ''
+                    ).ljust(3) + \
+                str('3') + \
+                str('0') + \
+                str('00') + \
+                str('00') + \
+                str('1') + \
+                str(invoice_line.product_id.intrastat_id.name or ''
+                    ).replace(' ', '')[0:8].ljust(8) + \
+                str('00') + \
+                str(sign_of_weight) + \
+                str(int(invoice_line.quantity * invoice_line.product_id.weight)
+                    ).replace('-', '').zfill(10) + \
+                str('+') + \
+                str('0000000000').zfill(10) + sign_of_invoice_value
+
+            if invoice_line.price_subtotal == 0.0:
+                standard_price = invoice_line.product_id.standard_price
+                quantity = invoice_line.quantity
+                value += str(int(standard_price * quantity)).replace(
+                    '-', '').zfill(10)
+            else:
+                value += str(int(invoice_line.price_subtotal)).replace(
+                    '-', '').zfill(10)
+            value += str('+') + str('0000000000').zfill(10)
+
+            if len(str(invoice_line.invoice_id.number or '')) < 8:
+                invoice_value = \
+                    str(invoice_line.invoice_id.number) + \
+                    str(invoice_line_number).zfill(2)
```

Comment:
You could use `(line_counter -1) % 100` so that you don't have to keep two counters. I find having two counters slightly confusing. But it's a matter of taste, of course (non blocking for merge).